### ISSUE TITLE
Null safety error after migration to 3.2.0

### ISSUE THREAD

**Issue by theochampion** [claimsforce-gmbh/flutter-segment#74]
Hey there,

I just migrated to `3.2.0` with init in Dart code and I now get this compile time error

```
Launching lib/main.dart on Pixel 4 in debug mode...
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment.dart:27:29: Error: The parameter 'options' can't have a value of 'null' because of its type 'SegmentConfig', but the implicit default value is 'null'.
 - 'SegmentConfig' is from 'package:flutter_segment/src/segment_config.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_config.dart').
Try adding either an explicit non-'null' default value or the 'required' modifier.
    @required SegmentConfig options,
                            ^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_config.dart:8:10: Error: The parameter 'writeKey' can't have a value of 'null' because of its type 'String', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    this.writeKey,
         ^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_config.dart:9:10: Error: The parameter 'trackApplicationLifecycleEvents' can't have a value of 'null' because of its type 'bool', but the implicit default value is 'null'.
Try adding either an explicit non-'null' default value or the 'required' modifier.
    this.trackApplicationLifecycleEvents,
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_platform_interface.dart:16:29: Error: The parameter 'options' can't have a value of 'null' because of its type 'SegmentConfig', but the implicit default value is 'null'.
 - 'SegmentConfig' is from 'package:flutter_segment/src/segment_config.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_config.dart').
Try adding either an explicit non-'null' default value or the 'required' modifier.
    @required SegmentConfig options,
                            ^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_method_channel.dart:10:29: Error: The parameter 'options' can't have a value of 'null' because of its type 'SegmentConfig', but the implicit default value is 'null'.
 - 'SegmentConfig' is from 'package:flutter_segment/src/segment_config.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_config.dart').
Try adding either an explicit non-'null' default value or the 'required' modifier.
    @required SegmentConfig options,
                            ^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_method_channel.dart:10:6: Error: Getter not found: 'required'.
    @required SegmentConfig options,
     ^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/lib/src/segment_method_channel.dart:10:6: Error: This can't be used as an annotation; an annotation should be a reference to a compile-time constant variable, or a call to a constant constructor.
    @required SegmentConfig options,
     ^


FAILURE: Build failed with an exception.
```

Any idea? 

Cheers!

**Comment by theochampion** [comment]
I was able to fix the issues by replacing instances of `@required` by the new `required` keyword and added default values in the SegmentConfig object :

```dart
  SegmentConfig({
    required this.writeKey,
    this.trackApplicationLifecycleEvents = false,
    this.amplitudeIntegrationEnabled = false,
    this.debug = false,
  });
```

**Comment by theochampion** [comment]
I've opened a pull request with the changes https://github.com/claimsforce-gmbh/flutter-segment/pull/76

**Comment by MaiKaY** [comment]
Should be solved with `3.2.1`
